### PR TITLE
Add MacDriver

### PR DESCRIPTION
**Pull request by penguinho**
## Proposed changes

Integrating the Mac OS X app driver

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x ] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [ x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ x] Lint and unit tests pass locally with my changes
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by imurchie** [comment]
Looks good to me! Two last things: 
1. can you add the driver to the name-driver mapping so we can log the version? https://github.com/penguinho/appium/blob/ef6e7a273124c24f9140456848e8510bc72cc5c0/lib/appium.js#L105
2. The driver needs to be published or this will break all Appium installs.

**Comment by penguinho** [comment]
@jlipps say the word and I shall merge, though I imagine we need to merge the base-driver changes first

**Review by imurchie — APPROVED** [review]

**Comment by jlipps** [comment]
i made a bunch of changes to appium-mac-driver that i'd like for us to review and merge and publish, then i think we're probably good to merge this. we should call out that this is beta only since it would be released potentially under 1.6.4 (or we could hold off till 1.7)

**Comment by jlipps** [comment]
Oh and I'd also want to see with this PR a doc file on how to get things set up (since you have to download a 3rd party binary and put it somewhere). users will no doubt want to know that!

**Comment by penguinho** [comment]
I've added the instructions. I would like to get this as a beta in 1.6.4 as I'm presenting about it in 2 weeks at SeConf in Austin and would love to make it easier for people to give it a try

File: README.md
Comment (jlipps) [inline README.md]: I think these instructions should go into the setup docs and not the main README. In this particular place I think it's too much detail for setup instructions, and people are more likely to confuse "mac requirements" with "ios requirements". I think mac app automation could have its own section in the osx setup doc file?

**Comment by penguinho** [comment]
@jlipps Done

**Comment by jlipps** [comment]
i squashed and merged, and will cherry-pick into the 1.6 branch

**Comment by jlipps** [comment]
ok, 1.6 branch is up to date, we'd just need to publish a new 1.6.4 beta with a different version number, i'll leave it up to @imurchie if we want to do that now or just get out 1.6.4 soon. but if you run appium from master, the mac driver is now present.

**Pull request merged**
